### ISSUE TITLE
ExecuteUpdate fails when target entity has an owned entity

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#28727]

**Comment by keatkeat87** [comment]
![image](https://user-images.githubusercontent.com/8691013/192242797-fd288f88-a8e6-466e-9682-76c1f2b1ebe6.png)
I got this error when calling ExecuteDeleteAsync on a model having Owned Entity.
https://github.com/keatkeat87/EFCore7_ExecuteDeleteAsync

**Comment by ajcvickers** [comment]
@keatkeat87 Please try with the [daily build](https://github.com/dotnet/efcore/blob/main/docs/DailyBuilds.md).

**Comment by Sebazzz** [comment]
I also got an issue with an entity that _has_ owned entities, but I'm setting the property on the entity itself:

```C#
            await this._applicationDbContext.Sources
                      .Where(s => s.CopiedFromId == source.Id)
                      .ExecuteUpdateAsync(action => action.SetProperty(s => s.CopiedFrom, (Source) null), cancellationToken);
```

It goes wrong here because `expression.Expression` is an `IncludeExpression` instead of `EntityShaperExpression`:
![image](https://user-images.githubusercontent.com/1426097/196364196-cacc2170-79be-481b-9421-c1a933f32902.png)
![image](https://user-images.githubusercontent.com/1426097/196364234-68b00376-acd1-4afa-a17f-b42e7cccb199.png)

This translation happens here, in the `ReplacingExpressionVisitor`:

![image](https://user-images.githubusercontent.com/1426097/196364444-5824c3ca-4590-42a7-8741-b28a71fdbeda.png)

**Comment by roji** [comment]
This happens because the property selector lambda has an IncludeExpression wrapping the RelationalEntityShaperExpression.

To unblock use of ExecuteUpdate on non-owned properties, we can just unwrap the IncludeExpressions (there may be more than one); as long as after unwrapping, we get a PropertyExpression directly over RelationalEntityShaperExpression, things should work fine. I will do this in #29672, which is very similar, allowing interface up-casts in update property selectors.

However, allowing ExecuteUpdate to reference **owned** properties is another matter (i.e. with table sharing, or possibly JSON - see #28766), and more complex. We should split that out to another issue.

**Comment by roji** [comment]
Allowing referencing owned properties in the property selector is covered by #28627.

**Comment by roji** [comment]
Clearing milestone to discuss patching
